### DISCUSSION TITLE
Using cx_freeze to create a library to run in process with another

### DISCUSSION THREAD

**Discussion by EmersonDove** [marcelotduarte/cx_Freeze#2290] [Q&A]
Hey everyone, been loving this library. We've been running a python service as a subprocess via a cx_Freeze executable, but just to simplify our stack we've been looking into embedding python into the parent process as a library.

Because `cx_Freeze` effectively creates an `int main` entrypoint, could the build system be modified to produce a dylib/dll instead of an executable with for example a `start` function?

https://github.com/marcelotduarte/cx_Freeze/blob/bfd4627a41025ce1206f416b90c5e41a46931e8c/source/bases/console.c#L47-L75

I just want to ask before embarking on changing the build system if there this approach is fundamentally flawed or if there are any big issues hiding anywhere.

I couldn't find anything in the docs or other discussions about this but perhaps I totally missed something!

**Comment by marcelotduarte** [comment]
Theoretically yes, you can make a new base and a new initscript. What is done in `console.c` is exactly embedding Python. I believe someone has already done this because in [initscripts](https://github.com/marcelotduarte/cx_Freeze/tree/main/cx_Freeze/initscripts) there are some files related to this, but I have no idea if it was just ideas.
